lib/route/link.c

```diff
@@ -2273,6 +2284,17 @@ uint8_t rtnl_link_get_carrier(struct rtnl_link *link)
 }
 
 /**
+ * Return carrier on/off changes of link object
+ * @arg link		Link object
+ *
+ * @return Carrier changes.
+ */
+uint32_t rtnl_link_get_carrier_changes(struct rtnl_link *link)
```

Comment:
libnl is not very consistent about this, but I think that in most cases getters should return an `int` value indicating success (negative values meaning error, `NLE_*`). Something like https://github.com/thom311/libnl/blob/993a62d4501a9b40461d9f1276a63246268b4a3a/lib/route/link/ppp.c#L197

Not sure about getters that return pointers like `rtnl_link_get_phys_switch_id`. But for getters that return integers, I think we shouldn't mix an error result with the value.